app/services/event_log/journey_runner.rb

```diff
@@ -0,0 +1,38 @@
+module EventLog
+  class JourneyRunner
+    attr_reader :journey
+
+    def initialize(journey)
+      @journey = journey
+    end
+
+    # Process events in order of client_timestamp
+    def call
+      # iterate over all events in the log and apply changes to the journey
+      events.each do |event| # NB: do not use events.find_each as it will break the ordering
+        case event.event_name
+        when Event::CANCEL, Event::COMPLETE, Event::REJECT, Event::START, Event::UNCANCEL, Event::UNCOMPLETE
+          # with a state-changing event, delegate it to the state machine
+          journey.send(event.event_name)
+        when Event::LOCKOUT, Event::LODGING
+          # no action to perform when a journey is locked out or lodged, these events are purely for auditing
+          # TODO: handle other journey events here
+        end
+      end
+
+      # save the journey if it has changed
+      if journey.changed? && journey.valid?
```

Comment:
I think you could rewrite these few lines as

```ruby
journey.save! if journey.changed? && journey.valid?
```

and have the same boolean result returned. `save!` returns `true` if no exception is raised, and `journey.changed? && journey.valid?` returns `false` unless both conditions are `true`.